elastic/kibana
# [Response Ops][Alerting] Creating global service for fetching and caching rules settings

Resolves https://github.com/elastic/kibana/issues/184321 and https://github.com/elastic/kibana/issues/149884

## Summary

Created a `RulesSettingsService` that caches flapping and query delay settings per space for 1 minute so rules are not accessing the SO index with every execution.

Also moved all classes related to the rules settings into the `rules_settings` folder so some of these changes are just renaming changes.

## To Verify
1. Add some logging to `x-pack/plugins/alerting/server/rules_settings/rules_settings_service.ts` to indicate when settings are being fetched and when they're returning from the cache.
2. Create and run some rules in different spaces to see that flapping and query delay settings are being returned from the cache when the cache has not expired.
